static_frame/core/frame.py

```diff
@@ -6657,68 +6657,78 @@ def unset_index(self, *,
             consolidate_blocks:
             columns_constructors:
         '''
+        # disallows specifying names with 'drop=True'
+        if drop and names:
+            raise RuntimeError("Cannot specify `names` when `drop=True`, as the index will not be added back as columns.")
+
         def blocks() -> tp.Iterator[TNDArrayAny]:
             # yield index as columns, then remaining blocks currently in Frame
-            if self._index.ndim == 1:
-                yield self._index.values
-            else:
-                # No recache is needed as it's not possible for an index to be GO
-                yield from self._index._blocks._blocks # type: ignore
-            for b in self._blocks._blocks:
-                yield b
+            if not drop:
+                if self._index.ndim == 1:
+                    yield self._index.values
+                else:
+                    # No recache is needed as it's not possible for an index to be GO
+                    yield from self._index._blocks._blocks # type: ignore
+            yield from self._blocks._blocks
 
         block_gen: tp.Callable[[], tp.Iterator[TNDArrayAny]]
         if consolidate_blocks:
             block_gen = lambda: TypeBlocks.consolidate_blocks(blocks())
         else:
             block_gen = blocks
 
-        if not names:
-            names = self._index.names
+        # if not names:
+        #     names = self._index.names
+        if drop:
+            # When dropping the index, keep the existing columns without adding index names
+            columns, own_columns = self._columns, True
```

Comment:
We can only own self._columns if this is not a `FrameGO`, i.e., it is purely immutable. We can substitute, the `STATIC` Boolean attr:
```suggestion
            columns, own_columns = self._columns, self.STATIC
```
Please add a test that specifically creates a `FrameGO`, calls this method, then grows (mutating columns) and then ensures the `Frame` resulting from this call does not have its columns mutated.

Reply:
Roger that.